integreat_cms/deepl_api/utils.py

```diff
@@ -110,6 +112,24 @@ def deepl_translation(self, request, content_objects, language_slug, form_class)
             # Get target language
             target_language = region.get_language_or_404(language_slug)
             source_language = region.get_source_language(language_slug)
+
+            if not mt_is_permitted(
+                region,
+                request.user,
+                type(content_objects.first())._meta.default_related_name,
+                language_slug,
+            ):
+                messages.error(
+                    request,
+                    _(
+                        'Machine translations are disabled for content type "{}" or for language "{}".'
+                    ).format(
+                        type(content_objects.first())._meta.verbose_name.title(),
```

Comment:
```suggestion
            if content_objects and not mt_is_permitted(
                region,
                request.user,
                type(content_objects[0])._meta.default_related_name,
                language_slug,
            ):
                messages.error(
                    request,
                    _(
                        'Machine translations are disabled for content type "{}", language "{}" or the current user.'
                    ).format(
                        type(content_objects[0])._meta.verbose_name.title(),
```
I see three potential problems:
1. When `content_objects` is empty, this will raise an `AttributeError: type object 'NoneType' has no attribute '_meta'`
2. The message is incorrect since the current user could also be the reason for the denied check
3. Calling `content_objects.first()` on a queryset you're going to need afterwards is inefficient, since it causes a separate db query to run. It's better to directly force the evaluation of the whole queryset (also using one query, but this time the result can be used afterwards when the whole queryset is iterated).

Reply:
1. yep, good point, thanks for fixing this.
2. thanks for the more complete error message (the chances of this being shown because of user permission issues are probably very low for legitimate use cases though)
3. did not know, TIL